vms/docker/start-server.sh

```diff
@@ -1,19 +1,43 @@
 #!/bin/sh
 
+# Generate SSL certificate if it doesn't exist
 if [ ! -f /standardebooks.org/web/config/ssl/standardebooks.test.crt ]; then
 	openssl req -x509 -nodes -days 99999 -newkey rsa:4096 -subj "/CN=standardebooks.test" -keyout /standardebooks.org/web/config/ssl/standardebooks.test.key -sha256 -out /standardebooks.org/web/config/ssl/standardebooks.test.crt
 fi
 
 cd /standardebooks.org/web
+
+# Update to required PHP version here if necessary
+
+# Run Composer install.
+# Note that the PHP version should be compatible with the packages in the composer.lock file.
 composer install
 
-ln -s /standardebooks.org/web/config/apache/standardebooks.test.conf /etc/apache2/sites-available/
-ln -s /standardebooks.org/web/config/php/fpm/standardebooks.org.ini /etc/php/*/cli/conf.d/
-ln -s /standardebooks.org/web/config/php/fpm/standardebooks.org.ini /etc/php/*/fpm/conf.d/
-ln -s /standardebooks.org/web/config/php/fpm/standardebooks.test.conf /etc/php/*/fpm/pool.d/
-a2ensite standardebooks.test
-service apache2 start
-service php7.4-fpm restart
+# Create symlink after checking if it already exists or removing the existing one
+create_symlink() {
+	target=$1
+	link=$2
+
+	if [ -L "$link" ]; then # Check if symlink already exists.
+		echo "Symlink $link already exists; skipping."
+	else
+		ln -s "$target" "$link"
+	fi
+}
+
+create_symlink /standardebooks.org/web/config/apache/standardebooks.test.conf /etc/apache2/sites-available/standardebooks.test.conf
+create_symlink /standardebooks.org/web/config/php/fpm/standardebooks.org.ini /etc/php/8.1/cli/conf.d/standardebooks.org.ini
+create_symlink /standardebooks.org/web/config/php/fpm/standardebooks.org.ini /etc/php/8.1/fpm/conf.d/standardebooks.org.ini
+create_symlink /standardebooks.org/web/config/php/fpm/standardebooks.test.conf /etc/php/8.1/fpm/pool.d/standardebooks.test.conf
+
+# Enable site only if it's not already enabled
```

Comment:
Do we really need to check if the site is already enabled, and if the services are already started? I thought this script ran once on container initialization. Does it actually run more than once?

Reply:
Nope, it only runs once if all is good.
I had to run the script multiple times while troubleshooting, that's why i left it and [this one as well](https://github.com/standardebooks/web/pull/300/files#diff-b93d6ca6f7ccd12896996adfb1f7adfbefea8c59f3d5fbaec870e85515832a56R21).

Reply:
OK, so let's leave those checks out, and also the checks on `service` and the `create_symlink` functino, because it's more code. We know this will all succeed because we're setting up the environment for that.

Reply:
Oh, actually - i need to correct myself.
The `ENTRYPOINT` script isn't just for the first `docker run`. But also for every time the container is restarted with `docker start`.

> The `ENTRYPOINT` specifies a command that will always be executed when the container starts.

https://stackoverflow.com/a/34245657